### ISSUE TITLE
Negative losses

### ISSUE THREAD

**Issue by arrayslayer** [sdv-dev/CTGAN#28]
* CTGAN version: 0.2.1
* Python version: 3.8
* Operating System: MacOS

### Description

While fitting and training, I get negative losses for Generator & Discriminator. What do negative losses imply? Shouldn't they be >=0?

### What I Did

Running a simple fit and train on adults dataset

**Comment by arrayslayer** [comment]
Also, can you tell which seed can we set to get reproducible samples?

**Comment by k15z** [comment]
Here are the two loss functions:

> loss_d = -(torch.mean(y_real) - torch.mean(y_fake))
> loss_g = -torch.mean(y_fake) + cross_entropy

The generator loss consists of two components, the `cross_entropy` which can never be negative and `-torch.mean(y_fake)` which can be negative if the discriminator assigns a high score to the fake/synthetic data.

Assuming that the discriminator has assigned a high score to the fake/synthetic data, then the only way that the discriminator loss could be negative is if it assigns an even higher score to the real data. 

Therefore, this suggests that the discriminator is, on average, assigning higher scores to real data than fake data which means it's doing its job.

Without more information, I don't see anything to be concerned about here.

**Comment by k15z** [comment]
Re: the random seed, this library use PyTorch so should be able to just add:

```
import torch
torch.manual_seed(42)
```

before you call `sample` to get reproducible samples.

**Comment by arrayslayer** [comment]
Thank you so much for both the detailed responses. Closing issue now :)
